src/test/java/org/minions/devfund/danielmontecinos/moviedatabase/MovieDatabaseTest.java

```diff
@@ -0,0 +1,168 @@
+        final int expectedSize = 1;
+
+        assertEquals("Movie 1", movieDatabase.getMovieList().get(0).getName());
+        assertEquals(expectedSize, movieDatabase.getMovieList().size());
+
+        assertEquals("Actor 1", movieDatabase.getMovieList().get(0).getActors().get(0).getName());
+        assertEquals("Actor 2", movieDatabase.getMovieList().get(0).getActors().get(1).getName());
+        assertEquals("Actor 3", movieDatabase.getMovieList().get(0).getActors().get(2).getName());
+    }
+
+    /**
+     * Tests that addMovie method doesn't add an exiting movie to movieList.
+     */
+    @Test
+    public void testAddMovieMethodDoesNotAddSameMovies() {
+        movieDatabase.addMovie("Movie 1", new String[]{"Actor 1", "Actor 2", "Actor 3"});
+        movieDatabase.addMovie("Movie 1", new String[]{"Actor 4", "Actor 5", "Actor 6"});
+
+        final int expectedSize = 1;
+        assertEquals(expectedSize, movieDatabase.getMovieList().size());
+    }
+
+    /**
+     * Tests that addMovie method can add different movies to movieList.
+     */
+    @Test
+    public void testAddMovieMethodAddsDifferentMovies() {
+        movieDatabase.addMovie("Movie 4", new String[]{"Actor 7"});
+        movieDatabase.addMovie("Movie 5", new String[]{"Actor 7", "Actor 9"});
+
+        final int expectedSize = 2;
+        assertEquals(expectedSize, movieDatabase.getMovieList().size());
+    }
+
+    /**
+     * Tests addMovie method only adds new actors to actorList.
+     */
+    @Test
+    public void testAddMovieMethodOnlyAddNewActorsToActorList() {
+        movieDatabase.addMovie("Movie 1", new String[] {"Actor 1", "Actor 2", "Actor 1", "Actor 2"});
+
+        final int expectedSize = 2;
+        assertEquals(expectedSize, movieDatabase.getActorList().size());
+    }
+
+    /**
+     * Tests addMovie method doesn't add existing actors to actorList.
+     */
+    @Test
+    public void testAddMovieMethodDoesNotAddSameActorsToActorList() {
+        movieDatabase.addMovie("Movie 9", new String[]{"Actor 1", "Actor 2"});
+        movieDatabase.addMovie("Movie 7", new String[]{"Actor 1", "Actor 2"});
+
+        final int expectedSize = 2;
+        assertEquals(expectedSize, movieDatabase.getActorList().size());
+    }
+
+    /**
+     * Tests addRating method adds a rating value to movie instance.
+     */
+    @Test
+    public void testAddRatingMethodAddsMovieRating() {
+        final double actualRating = 98.0;
+        movieDatabase.addMovie("Movie 7", new String[] {"Actor 6"});
+        movieDatabase.addRating("Movie 7", actualRating);
+
+        final double expectedRating = 98.0;
+        assertEquals(expectedRating, movieDatabase.getMovieList().get(0).getRating(), 0.0);
+    }
+
+    /**
+     * Tests that updateRating methods updates rating previously set.
+     */
+    @Test
+    public void testUpdateRatingMethodUpdatesMovieRatingSetsBeforeUseIt() {
+        movieDatabase.addMovie("Movie 10", new String[] {"Actor 10"});
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Define a constant instead of duplicating this literal "Movie 10" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)